### PR TITLE
Match `func_801457CC` (z_sram_NES.c OK)

### PR DESCRIPTION
**Pull request by engineer124**

### CODE REVIEW

**Review by hensldm — APPROVED** [review]

File: src/code/z_sram_NES.c [hunk 1]
```diff
@@ -1424,23 +1423,23 @@ void func_801457CC(GameState* gameState, SramContext* sramCtx) {
...
                         if ((gSaveContext.save.saveInfo.playerData.newf[0] == 'Z') &&
-                            (gSaveContext.save.saveInfo.playerData.newf[1] == 'E')) {
-                            phi_s2 = false;
-                        }
-
-                        // phi_s2 = false;
+                            (gSaveContext.save.saveInfo.playerData.newf[1] == 'E') &&
+                            (gSaveContext.save.saveInfo.playerData.newf[2] == 'L') &&
+                            (gSaveContext.save.saveInfo.playerData.newf[3] == 'D') &&
+                            (gSaveContext.save.saveInfo.playerData.newf[4] == 'A')) {}
```
Comment (AngheloAlf) [inline src/code/z_sram_NES.c:1439]: Of course. Why wouldn't it be like this? :upside_down_face:

Comment (AngheloAlf) [reply src/code/z_sram_NES.c:1439]: Memes aside, does the `CHECK_NEWF` macro works here?

File: src/code/z_sram_NES.c [hunk 2]
```diff
@@ -1490,14 +1487,14 @@ void func_801457CC(GameState* gameState, SramContext* sramCtx) {
                     fileSelect->rupees[sp76] = gSaveContext.save.saveInfo.playerData.rupees;
                     fileSelect->walletUpgrades[sp76] = CUR_UPG_VALUE(UPG_WALLET);
 
-                    for (sp7A = 0, phi_a0 = 0; sp7A < 24; sp7A++) {
-                        if (gSaveContext.save.saveInfo.inventory.items[sp7A + 24] != 0xFF) {
-                            phi_a0++;
+                    for (sp7A = 0, maskCount = 0; sp7A < 24; sp7A++) {
+                        if (gSaveContext.save.saveInfo.inventory.items[sp7A + 24] == ITEM_NONE) {
```
Comment (AngheloAlf) [inline src/code/z_sram_NES.c:1491]: What do you think about this? I'm not totally convinced by it
```suggestion
                        if (gSaveContext.save.saveInfo.inventory.items[sp7A + SLOT_MASK_POSTMAN] == ITEM_NONE) {
```

This loop seems to just count how many masks the player has. I guess we could introduce stuff like `SLOT_MASK_MIN` and `SLOT_MASK_MAX`, but that is outside of the scope of this PR...

Feel free to ignore this whole suggestion lol, just throwing some thoughts.

File: src/code/z_sram_NES.c [hunk 3]
```diff
@@ -1539,24 +1536,26 @@ void func_801457CC(GameState* gameState, SramContext* sramCtx) {
...
                             // checksum didnt match, try backup save
                             sp6E = 1;
                             if ((gSaveContext.save.saveInfo.playerData.newf[0] == 'Z') &&
-                                (gSaveContext.save.saveInfo.playerData.newf[1] == 'E')) {
-                                phi_s2 = false;
-                            }
+                                (gSaveContext.save.saveInfo.playerData.newf[1] == 'E') &&
+                                (gSaveContext.save.saveInfo.playerData.newf[2] == 'L') &&
+                                (gSaveContext.save.saveInfo.playerData.newf[3] == 'D') &&
+                                (gSaveContext.save.saveInfo.playerData.newf[4] == 'A')) {}
```
Comment (AngheloAlf) [inline src/code/z_sram_NES.c:1553]: Same question about `CHECK_NEWF`

**Review by AngheloAlf — APPROVED** [review]

File: src/code/z_sram_NES.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_sram_NES.c:1439]: It does not, I tried.

File: src/code/z_sram_NES.c (hunk 2 above)
Comment (engineer124) [reply src/code/z_sram_NES.c]: A macro already exists for this an was used in kaleido: `ITEM_NUM_SLOTS`.

I've gone ahead and used it here and in other places

**Review by AngheloAlf — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

**Pull request merged**